woocommerce/woocommerce-ios
# REST API: Custom cookie nonce authenticator

Part of #8453 


## Description

There has been an ongoing issue with handling cookie-nonce authentication for Pressable sites (also mentioned by @joshheald  in #8587). This blocks testing for these sites and can be an issue on production as well (though not reproducible or reported yet).

Previously we were using `WordPressKit`'s `CookieNonceAuthenticator` to handle the authentication, but there's a potential issue with using the cookie nonce retrieval URL as the redirect URL for the login request. 

This PR creates an updated authenticator that handles nonce retrieval as a separate request. This seems to fix the failures on Pressable sites. I'm not creating a PR to `WordPressKit` to avoid affecting other apps that are using this library.

Also: in order to avoid confusion, this PR also renames the existing `RequestAuthenticator`, `RequestProcessor` with the `ApplicationPassword` prefix.

Due to time constraints, unit tests for the authenticator will be added in a future PR.

## Testing instructions

Pre-requisite: Make sure that you have a test site hosted on Pressable.
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app or skip onboarding if needed.
- On the prologue screen, select Enter your site address and proceed with the address of your test site.
- Log in with your site credentials. Notice that the login succeeds.
- Please feel free to try a few times again to make sure that the authentication succeeds in case the failure is not consistent.

## Screenshots

N/A

---
